File: domain/src/main/java/com/thinkzi/weather/domain/usecase/base/ObservableObserver.java
package com.thinkzi.weather.domain.usecase.base;

import io.reactivex.observers.DisposableObserver;

/**
 * provide a observer for the work result (T object or complete or error)
 * */
public class ObservableObserver<T> extends DisposableObserver<T> {

    @Override
    public void onNext(T t) {

    }

    @Override
    public void onError(Throwable e) {

    }

    @Override
    public void onComplete() {

    }

}

File: domain/src/main/java/com/thinkzi/weather/domain/usecase/base/ObservableObserver.kt
package com.thinkzi.weather.domain.usecase.base

import io.reactivex.observers.DisposableObserver

/**
 * provide a observer for the work result (T object or complete or error)
 */
class ObservableObserver<T> : DisposableObserver<T>() {

    override fun onNext(t: T) {

    }

    override fun onError(e: Throwable) {

    }

    override fun onComplete() {

    }

}
